pgcli/main.py

```diff
@@ -383,6 +379,18 @@ def initialize_logging(self):
         pgspecial_logger.addHandler(handler)
         pgspecial_logger.setLevel(log_level)
 
+    def initialize_keyring(self):
+        self.keyring = None
+
+        keyring_enabled = self.config["main"].as_bool("keyring")
+        if keyring_enabled:
+            # Try best to load keyring (issue #1041).
+            import importlib
+            try:
+                self.keyring = importlib.import_module('keyring')
```

Comment:
@zzl0 What do you think about wrapping this call into a timeout decorator? Something like described here:

https://stackoverflow.com/questions/2281850/timeout-function-if-it-takes-too-long-to-finish

for those cases when it takes a long time to fail, or just takes too long.

Reply:
NOTE: I assume the root cause is keyring has a bug/feature.

If a user's keyring doesn't work properly, then the user probably has 3 options:

1. fix keyring, make it work properly.
2. turn off keyring in the config file.
3. don't fix keyring, pgcli wait for some time and then continue without keyring, the effect is same as option 2 and user need to wait for some time.

So I prefer not to do it. What do you think?

Reply:
I was thinking of preventing the user from waiting too long in case of 3, and getting frustrated with pgcli. I suppose it would only happen once, after which we could tell the user to disable keyring. It's just that not every user will be able to figure out that keyring is the slow-loading part. It is likely that they will blame pgcli, and perhaps give up on pgcli.

This is still a good fix, and I'm going to merge it for now.

Reply:
Thanks for sharing your thought. Yeah, I agree with that. But I was also worry that users might think pgcli startup is slow (if their keyring always timeout and they don't check the log).

Let's see if users have the problem that you mentioned, we can fix it then.